### PR TITLE
HAI-865 add pyenv and screen to base img

### PR DESCRIPTION
**Pull request by arunkumarpandian**

### CODE REVIEW

File: base/Dockerfile
Comment (arunkumarpandian) [inline base/Dockerfile]: Adding these because `pyenv install <version>` would fail without these dependencies.

```
Installing Python-3.8.15...
WARNING: The Python bz2 extension was not compiled. Missing the bzip2 lib?
WARNING: The Python curses extension was not compiled. Missing the ncurses lib?
WARNING: The Python ctypes extension was not compiled. Missing the libffi lib?
WARNING: The Python readline extension was not compiled. Missing the GNU readline lib?
ERROR: The Python ssl extension was not compiled. Missing the OpenSSL lib?
```

File: base/Dockerfile
Comment (anaoum) [inline base/Dockerfile]: I’d make this an ARG not ENV

potentially true for PYENV_ROOT also. Is it a variable required only at install time or when running the container too?

Comment (dcarrion87) [reply base/Dockerfile]: Also just be mindful of notifying staff before this is merged in. There's a chance people have done their own thing with pyenv. E.g. I have. I'm sure staff will be fine but just something to be mindful if it's going to override their environment.

Comment (arunkumarpandian) [reply base/Dockerfile]: @dcarrion87 Could you let me know what is different/custom in your setup ? @anaoum Good point. I think PYENV_GIT_TAG could be made as an ARG. Shall double check on PYENV_ROOT to see if it isn't needed at run-time.

Comment (dcarrion87) [reply base/Dockerfile]: It's just the standard set up by the looks with this in my .bashrc:

```
export PYENV_ROOT="$HOME/.pyenv"
command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init -)"
```

Adding pyenv in docker container is actually going to be negated since `/home/coder` mounts into a unique user NFS folder on start.

Given that pyenv does so much in the users home folder I think just put the dependencies in and let the user manage their own pyenv underneath their mounted `/home/coder/.pyenv`.

Otherwise we need to put pyenv somewhere else and I'm not sure we want to intervene with users on providing a global pyenv? Could get messy and they might not like having their pyenv path overwritten by default if they have customisations.

E.g.: Dear staff, dependencies for pyenv are now in the container. If you want it run these one liners to add it into your home folder: ...

If we do want to deliver it automatically it's probably a post hook start up script that first checks if `$HOME/.pyenv` exists first and then does the thing. But not sure if it's worth given it's a one off thing.

Comment (arunkumarpandian) [reply base/Dockerfile]: @anaoum & @andychisholm could you let me know if your preference would also be to leave out pyenv and just install its dependencies as recommended by @dcarrion87 ?

Comment (andychisholm) [reply base/Dockerfile]: @arunkumarpandian I agree -- this feels like an analogous situation to direnv.

Let's "support" it with all the packages it needs but not actually install/configure it for users; given it's happy install path seems to be at odds with our persistence pattern.

File: base/Dockerfile
Comment (andychisholm) [reply base/Dockerfile]: Wonder if we should comment on the block to explain the "upstream" reason for all these packages, so we don't regress support in the future.

Also wonder if there's some other package that hits all these already?

**Review by andychisholm — APPROVED** [review]
Approved, assuming we will remove the pyenv install from the dockerfile per https://github.com/harrison-ai/cobalt-docker-coder-dev/pull/5/files#r1047862551

**Review by jamesconstable — APPROVED** [review]

File: base/Dockerfile
Comment (arunkumarpandian) [reply base/Dockerfile]: OK. shall just add the dependencies into the image.

File: base/Dockerfile
Comment (arunkumarpandian) [reply base/Dockerfile]: Have added a comment now. And have filtered any redundant entries from the install list.

**Pull request merged**
